target/scripts/startup/setup-stack.sh

```diff
@@ -345,28 +346,98 @@ function _setup_dovecot_local_user
-      # user:password:uid:gid:(gecos):home:(shell):extra_fields
-      # Example :
-      # ${LOGIN}:${PASS}:5000:5000::/var/mail/${DOMAIN}/${USER}::userdb_mail=maildir:/var/mail/${DOMAIN}/${USER}
-      echo "${LOGIN}:${PASS}:5000:5000::/var/mail/${DOMAIN}/${USER}::${USER_ATTRIBUTES}" >> /etc/dovecot/userdb
+      # Dovecot's userdb has the following format
+      #   user:password:uid:gid:(gecos):home:(shell):extra_fields
+      echo \
+        "${LOGIN}:${PASS}:5000:5000::/var/mail/${DOMAIN}/${USER}::${USER_ATTRIBUTES}" \
+        >>/etc/dovecot/userdb
+
       mkdir -p "/var/mail/${DOMAIN}/${USER}"
 
-      # Copy user provided sieve file, if present
+      # copy user provided sieve file, if present
       if [[ -e "/tmp/docker-mailserver/${LOGIN}.dovecot.sieve" ]]
       then
         cp "/tmp/docker-mailserver/${LOGIN}.dovecot.sieve" "/var/mail/${DOMAIN}/${USER}/.dovecot.sieve"
       fi
 
       echo "${DOMAIN}" >> /tmp/vhost.tmp
     done < <(grep -v "^\s*$\|^\s*\#" /tmp/docker-mailserver/postfix-accounts.cf)
+
+    # see https://github.com/docker-mailserver/docker-mailserver/pull/2248#issuecomment-953313852
+    # for more details on this section
+    if [[ -f /tmp/docker-mailserver/postfix-virtual.cf ]] && [[ ${ENABLE_QUOTAS} -eq 1 ]]
+    then
+      # adding aliases to Dovecot's userdb
+      # ${REAL_FQUN} is a user's fully-qualified username
+      local ALIAS REAL_FQUN
+      while read -r ALIAS REAL_FQUN
+      do
+        # ignore comments
+        [[ ${ALIAS} == \#* ]] && continue
+
+        # alias is assumed to not be a proper e-mail
+        # these aliases do not need to be added to Dovecot's userdb
+        [[ ! ${ALIAS} == *@* ]] && continue
+
+        # clear possibly already filled arrays
+        unset REAL_ACC USER_QUOTA
+        declare -a REAL_ACC USER_QUOTA
+
+        local REAL_USERNAME REAL_DOMAINNAME
+        REAL_USERNAME=$(cut -d @ -f1 <<< "${REAL_FQUN}")
+        REAL_DOMAINNAME=$(cut -d @ -f2 <<< "${REAL_FQUN}")
+
+        if ! grep -q "${REAL_FQUN}" /tmp/docker-mailserver/postfix-accounts.cf
+        then
+          _notify 'inf' "Alias '${ALIAS}' is non-local (or mapped to a non-existing account) and will not be added to Dovecot's userdb"
+          continue
+        fi
+
+        _notify 'inf' "Adding alias '${ALIAS}' for user '${REAL_FQUN}' to Dovecot's userdb"
+
+        # REAL_ACC is an indexed array with
+        #
+        # | position | content                                       |
+        # |        0 | real account name (e-mail address) == ${FQUN} |
+        # |        1 | password hash                                 |
+        # |        2 | optional user attributes                      |
+        IFS='|' read -r -a REAL_ACC < <(grep "${REAL_FQUN}" /tmp/docker-mailserver/postfix-accounts.cf)
+
+        if [[ -z ${REAL_ACC[1]} ]]
+        then
+          _notify 'err' "Could not acquire password hash of user '${REAL_FQUN}' while adding alias '${ALIAS}'"
```

Comment:
What is the difference between `err` and `fatal` if not killing `docker-mailserver`?

This will be run during `check-for-changes.sh` as well. If `fatal` would imply bringing down the container it probably shouldn't happen during change events :sweat_smile: (then again, that's probably going to be the case when it starts calling the `SSL_TYPE` handling methods which call `dms_panic` for some failures.

Reply:
Wrong word choice. I meant something fatal like dms_panic.

Reply:
Hmm, I see. On the one hand, one could argue that shutting was too hard here, but then, this error also indicates that something is wrong in `/tmp/docker-mailserver/postfix-accounts.cf`, which is fatal, I agree. I will tend to it.